pkg/arvo/app/chat.hoon

```diff
@@ -225,109 +164,82 @@
-            |=  [shp=@p stat=status:hall]
-            shp
-          (~(put by acc) cir newset)
-        ::
-        :-
-          %+  turn  ~(tap in (~(del in (silt circles)) [our.bol %inbox]))
-            |=  cir=circle:hall
-            ^-  move
-            =/  wir/wire  /circle/[(scot %p our.bol)]/[nom.cir]/config/group
-            =/  pat/path  /circle/[nom.cir]/config/group
-            [ost.bol %peer wir [our.bol %hall] pat]
-          ::
-          %=  this
-              inbox.str.sta  loc.cos.piz
-              configs.str.sta  configs
-              messages.str.sta  (molt meslis)
-              peers.str.sta  (~(put by peers) [our.bol %inbox] localpeers)
-          ==
+    =/  peers=(map circle:hall (set @p))
+      %-  ~(rep by rem.pes.piz)
+      |=  [[cir=circle:hall grp=group:hall] acc=(map circle:hall (set @p))]
+      ^-  (map circle:hall (set @p))
+      (~(put by acc) cir (silt (turn ~(tap by grp) head)))
     ::
-    ::  fill remote configs with message data
+    :-
+      %+  turn  ~(tap in (~(del in (silt circles)) [our.bol %inbox]))
+      |=  cir=circle:hall
+      ^-  move
+      =/  wir/wire  /circle/[(scot %p our.bol)]/[nom.cir]/config/group
+      =/  pat/path  /circle/[nom.cir]/config/group
+      [ost.bol %peer wir [our.bol %hall] pat]
+    %=  this
+        inbox.str  loc.cos.piz
+        configs.str  configs
+        messages.str  (molt meslis)
+        peers.str  (~(put by peers) [our.bol %inbox] localpeers)
+    ==
+  ::
+  ::  fill remote configs with message data
+  ::
+    =*  messages  messages.str
+    =/  circle=circle:hall  [`@p`(slav %p &2:wir) &3:wir]
+    =/  localpeers=(set @p)  (silt (turn ~(tap by loc.pes.piz) head))
     ::
-      =*  messages  messages.str.sta
-      =/  circle/circle:hall  [`@p`(slav %p &2:wir) &3:wir]
-      =/  localpeers/(set @p)
-        %-  silt  %+  turn  ~(tap by loc.pes.piz)
-        |=  [shp=@p stat=status:hall]
-        shp
-      ::
-      =/  peers/(map circle:hall (set @p))
-        %-  ~(rep by rem.pes.piz)
-        |=  [[cir=circle:hall grp=group:hall] acc=(map circle:hall (set @p))]
-        ^-  (map circle:hall (set @p))
-        =/  newset
-          %-  silt  %+  turn  ~(tap by grp)
-          |=  [shp=@p stat=status:hall]
-          shp
-        (~(put by acc) cir newset)
-      =/  str
-        %=  str.sta
-          messages  (~(put by messages) circle nes.piz)
-          peers  (~(uni by peers.str.sta) (~(put by peers) circle localpeers))
-        ==
-      =/  messageupdate/update
-        :*  %messages
-            circle
-            0
-            (lent messages)
-            nes.piz
-        ==
-      :-  (send-chat-update [messageupdate str])
-      this(str.sta str)
+    =/  peers=(map circle:hall (set @p))
+      %-  ~(rep by rem.pes.piz)
+      |=  [[cir=circle:hall grp=group:hall] acc=(map circle:hall (set @p))]
+      ^-  (map circle:hall (set @p))
+      (~(put by acc) cir (silt (turn ~(tap by grp) head)))
+    =/  sta
+      %=  str
```

Comment:
You can do the `=.  str` here again.

Reply:
large enough statement that I prefer the %=

Reply:
Uh, I mean 

```hoon
=.  str
  %=  str
```

instead of with `=/  sta`.